trie/verkle.go

```diff
@@ -465,9 +455,14 @@ func (t *VerkleTrie) UpdateContractCode(addr common.Address, codeHash common.Has
 
 		// Reuse the calculated key to also update the code size.
 		if i == 0 {
-			cs := make([]byte, 32)
-			binary.LittleEndian.PutUint64(cs, uint64(len(code)))
-			values[utils.CodeSizeLeafKey] = cs
+			headervals, err := t.root.(*verkle.InternalNode).GetValuesAtStem(key[:31], t.FlatdbNodeResolver)
+			if err != nil {
+				return fmt.Errorf("UpdateContractCode (addr=%x) error while getting account header: %w", addr[:], err)
+			}
+			var cs [8]byte
+			binary.LittleEndian.PutUint64(cs[:], uint64(len(code)))
+			copy(headervals[utils.BasicDataLeafKey][utils.BasicDataCodeSizeOffset:], cs[5:8])
+			values[utils.BasicDataLeafKey] = headervals[utils.BasicDataLeafKey]
```

Comment:
This is the part I like the least: the simplicity we gained from using the basic data leaf, we recover in this particular place: now, in order to write the code size, we need to first load the leaf.

Until now, `UpdateStem` would go over all leaves and only overwrite the leaves that were `nil`. That used to be enough, because code size was in its own leaf. This is no longer the case, so the leaf has first to be fetched and then the byte range updated.

I think it can be done more efficiently, but I need to write it down. I will not merge this until I got to the bottom of the story.